rclpy/rclpy/node.py

```diff
@@ -39,26 +52,61 @@
 from rclpy.validate_namespace import validate_namespace
 from rclpy.validate_node_name import validate_node_name
 from rclpy.validate_topic_name import validate_topic_name
+from rclpy.waitable import Waitable
 
 HIDDEN_NODE_PREFIX = '_'
 
+# Used for documentation purposes only
+MsgType = TypeVar('MsgType')
+SrvType = TypeVar('SrvType')
+SrvTypeRequest = TypeVar('SrvTypeRequest')
+SrvTypeResponse = TypeVar('SrvTypeResponse')
+
 
 class Node:
+    """
+    A Node in the ROS graph.
+
+    A Node is the primary entrypoint in a ROS system for communication.
+    It can be used to create ROS entities such as publishers, subscribers, services, etc.
+    """
 
     def __init__(
-        self, node_name, *, context=None, cli_args=None, namespace=None, use_global_arguments=True,
-        start_parameter_services=True, initial_parameters=None
-    ):
+        self,
+        node_name: str,
+        *,
+        context: Context = None,
+        cli_args: List[str] = None,
+        namespace: str = None,
+        use_global_arguments: bool = True,
+        start_parameter_services: bool = True,
+        initial_parameters: List[Parameter] = None
+    ) -> None:
+        """
+        Constructor.
+
+        :param node_name: A name to give to this node. Validated by :func:`validate_node_name`.
+        :param context: The context to be associated with, or ``None`` for the default global
+            context.
+        :param cli_args: A list of strings of command line args to be used only by this node.
+        :param namespace: The namespace to which relative topic and service names will be prefixed.
+            Validated by :func:`validate_namespace`.
+        :param use_global_arguments: ``False`` if the node should ignore process-wide command line
+            args.
+        :param start_parameter_services: ``False`` if the node should not create parameter
+            services.
+        :param initial_parameters: A list of parameters to be set during node creation.
+        """
         self._handle = None
         self._context = get_default_context() if context is None else context
-        self._parameters = {}
-        self.publishers = []
-        self.subscriptions = []
-        self.clients = []
-        self.services = []
-        self.timers = []
-        self.guards = []
-        self.waitables = []
+        self._parameters = {}  # type: dict
+        self.publishers = []  # type: List[Publisher]
+        self.subscriptions = []  # type: List[Subscription]
+        self.clients = []  # type: List[Client]
+        self.services = []  # type: List[Service]
+        self.timers = []  # type: List[WallTimer]
+        self.guards = []  # type: List[GuardCondition]
+        self.waitables = []  # type: List[Waitable]
```

Comment:
Since we're no longer supporting Python <3.6, you can (and imo should) use the non-comment based syntax, e.g. `self.waitables: List[Waitable] = []`.

Reply:
Updated: 0cabd58